targets/PreChargeKEV1N/main.cpp

```diff
@@ -0,0 +1,209 @@
+/**
+* This is the primary State machine handler for the pre-charge voltage controller (PVC) board
+*/
+
+#include <EVT/io/CANopen.hpp>
+#include <EVT/io/UART.hpp>
+#include <EVT/io/pin.hpp>
+#include <EVT/io/types/CANMessage.hpp>
+#include <EVT/manager.hpp>
+#include <EVT/utils/log.hpp>
+#include <EVT/utils/types/FixedQueue.hpp>
+
+#include <PreCharge/GFDB.hpp>
+#include <PreCharge/PreChargeKEV1N.hpp>
+
+namespace IO = EVT::core::IO;
+namespace DEV = EVT::core::DEV;
+namespace time = EVT::core::time;
+
+///////////////////////////////////////////////////////////////////////////////
+// EVT-core CAN callback and CAN setup. This will include logic to set
+// aside CANopen messages into a specific queue
+///////////////////////////////////////////////////////////////////////////////
+
+struct CANInterruptParams {
+    EVT::core::types::FixedQueue<CANOPEN_QUEUE_SIZE, IO::CANMessage>* queue;
+    IO::CANf3xx* can;
+};
+
+/**
+* Interrupt handler to get CAN messages. A function pointer to this function
+* will be passed to the EVT-core CAN interface which will in turn call this
+* function each time a new CAN message comes in.
+*
+* NOTE: For this sample, every non-extended (so 11 bit CAN IDs) will be
+* assummed to be intended to be passed as a CANopen message.
+*
+* @param message[in] The passed in CAN message that was read.
+*/
+void canInterruptHandler(IO::CANMessage& message, void* priv) {
+    auto* canStruct = (struct CANInterruptParams*) priv;
+    if (message.isCANExtended()) {
+        if (canStruct != nullptr && canStruct->can != nullptr) {
+            canStruct->can->addCANMessage(message);
+        }
+    } else {
+        if (canStruct != nullptr && canStruct->queue != nullptr) {
+            canStruct->queue->append(message);
+        }
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// CANopen specific Callbacks. Need to be defined in some location
+///////////////////////////////////////////////////////////////////////////////
+extern "C" void CONodeFatalError(void) {
```

Comment:
This stuff will be unnecessary eventually with Taylor's CanOpen changes, but I assume it's all good for now.